cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -2,126 +2,21 @@ package cli
-	TelemetrySocket string        `optional:"" help:"Socket address (TCP/UNIX) to emit metrics to (e.g. 127.0.0.1:8200)"`
-}
-
-type CliContext struct {
-	JWTSourceProvider credentials.JWTSourceProvider
-	STSProvider       credentials.STSProvider
-	Logger            *logrus.Logger
-	Telemetry         *telemetry.Telemetry
-}
-
-func (c *CredentialsCmd) Run(context *CliContext) (err error) {
-	c.configureLogger(context.Logger)
-
-	t, err := c.configureTelemetry()
-	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to configure telemetry for socket address %s", c.TelemetrySocket))
-	}
-	defer t.Close()
-	context.Telemetry = t
-
-	emitMetrics := t.Instrument([]string{"Cli", "Run"}, &err)
-	defer emitMetrics()
-
-	spiffeID, err := spiffeid.FromString(c.SpiffeID)
-	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to parse SPIFFE ID from %s", c.SpiffeID))
-	}
-
-	src := context.JWTSourceProvider(spiffeID, c.WorkloadSocket, c.Audience, context.Logger, t)
-
-	session := createSession(c.STSEndpoint, c.STSRegion)
-	stsClient := context.STSProvider(session)
-
-	provider, err := credentials.NewProvider(
-		c.Audience,
-		c.RoleARN,
-		src,
-		c.SessionDuration,
-		stsClient,
-		t)
-	if err != nil {
-		return errors.Wrap(err, "failed to instantiate credentials provider")
-	}
-
-	creds, err := processcreds.SerializeCredentials(provider)
-	if err != nil {
-		return errors.Wrap(err, "failed to serialize credentials")
-	}
-
-	_, err = fmt.Print(string(creds))
-	return err
-}
-
-func (c *CredentialsCmd) configureLogger(logger *logrus.Logger) {
-	if len(c.LogFilePath) > 0 {
-		file, err := os.OpenFile(c.LogFilePath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
-		if err != nil {
-			logger.Info(errors.Wrapf(err, "Failed to log to file %s, using default of stderr", c.LogFilePath))
-		} else {
-			logger.Out = io.MultiWriter(os.Stderr, file)
-		}
-	}
-}
-
-func (c *CredentialsCmd) configureTelemetry() (t *telemetry.Telemetry, err error) {
-	t, err = telemetry.NewTelemetry(c.TelemetrySocket)
-	if err != nil && len(c.STSRegion) > 0 {
-		t.AddLabel("stsRegion", c.STSRegion)
-	}
-	return
-}
-
 type CLI struct {
 	Credentials CredentialsCmd `kong:"cmd,help:'print credentials in a format usable as an AWS credentials_process'"`
 }
 
 func RunWithDefaultContext(args []string) error {
-	logger := logrus.New()
-	// Example log line:
-	// time="2021-03-01T16:28:51-08:00" level=error msg="failed to parse SPIFFE ID from 305d07ed-765e-4642-9bd3-4c74aa86f5ed: spiffeid: invalid scheme"
-	logger.SetFormatter(&logrus.TextFormatter{})
-
-	telemetry, err := telemetry.NullTelemetry()
+	context, err := NewDefaultCliContext()
 	if err != nil {
 		return err
 	}
-
-	context := &CliContext{
-		JWTSourceProvider: credentials.StandardJWTSourceProvider,
-		STSProvider:       credentials.StandardSTSProvider,
-		Logger:            logger,
-		Telemetry:         telemetry,
-	}
-
 	return Run(context, args)
 }
```

Comment:
Now that we've got a helper method for creating the default CliContext, this method is almost empty.